client/iotcore/client.go

```diff
@@ -208,6 +208,14 @@ func (c *client) UpsertDevice(ctx context.Context,
 			})
 	}
 
+	endpointType := "iot:Data-ATS"
+	endpointOutput, err := svc.DescribeEndpoint(ctx, &iot.DescribeEndpointInput{
+		EndpointType: &endpointType,
+	})
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I don't think you need this at all...

Reply:
Shouldn't the fail-fast rule be used here? In case of error, we wouldn't have to check all `if err == nil` instructions.

Reply:
Besides, those `if err == nil` checks may cause ineffectual assignment static checker errors, like [here](https://gitlab.com/Northern.tech/Mender/iot-manager/-/jobs/2961868194#L37). Please let me know what do you think.

Reply:
it doesn't matter, make it pass the CI/CD pipeline and I'm okay.